enhancements/update/synchronized-upgrades.md

```diff
@@ -0,0 +1,302 @@
+  and/or workers without DPUs.)
+
+## Motivation
+
+### Goals
+
+- Make upgrades work smoothly in clusters running with DPU support, by
+  synchronizing the reboots of nodes between the infra cluster and the
+  tenant cluster.
+
+### Non-Goals
+
+- Supporting synchronized upgrades of more than 2 clusters at once.
+
+## Proposal
+
+### User Stories
+
+As the administrator of a cluster using DPUs, I want to be able to do
+y-stream and z-stream upgrades without causing unnecessary network outages.
+
+### API Extensions
+
+TBD
+
+### Implementation Details/Notes/Constraints
+
+#### Setup
+
+We can set some things up at install time (eg, creating credentials to
+allow certain operators in the two clusters to talk to each other).
+
+#### Initiating Upgrades
+
+As part of the DPU security model, the tenant cluster cannot have any
+power over the infra cluster. In particular, it can't be possible for
+an administrator in the tenant cluster to force the infra cluster to
+upgrade/downgrade to any particular version.
+
+Thus, the infra cluster upgrade must be initiated on the infra cluster
+side. In theory, we could have the infra cluster then initiate the
+tenant cluster upgrade, but that would require having a tenant cluster
+cluster-admin admin credential lying around in the infra cluster
+somewhat unnecessarily. It probably makes more sense to require the
+administrator to initiate the upgrade on the tenant side as well.
+
+#### Inter-Cluster Version Skew
+
+There is very little explicit communication between the two clusters:
+
+  1. The dpu-operator in the infra cluster talks to the apiserver in
+     the tenant cluster.
+
+  2. The ovn-kubernetes components in the two clusters do not
+     communicate directly, but they do make some assumptions about
+     each other's behavior (which is necessary to coordinate the
+     plumbing of pod networking).
+
+The apiserver communication is very ordinary and boring and not likely
+to be subject to any interesting version skew problems (even if we
+ended up with, say, a 3-minor-version skew between the clusters).
+
+Thus, other than ovn-kubernetes, no OCP component needs to be
+concerned about version skew between the two clusters, because the
+components in the two clusters are completely unaware of each other.
+
+For ovn-kubernetes, if we ever change the details of the cross-cluster
+communication, then we will need to add proper checks to enforce
+tolerable cross-cluster skew at that time.
+
+## Design Details
+
+The upgrade plan takes advantage of two things:
+
+- Other than the reboots, no part of the upgrade needs to be
+  synchronized between the two clusters.
+
+- Each infra node only runs pods to support the pods on its
+  corresponding tenant node. Thus, if a tenant node reboots, it is
+  safe to immediately reboot its infra node as well without needing
+  to cordon and drain it, because it is guaranteed that none of its
+  pods are doing anything important at that point.
```

Comment:
OK...I have a potentially dumb question - when the tenant reboots, the infra node can (by default) just continue running?  I would have thought that there'd be some platform initialization (e.g. of the PCI bus) that might break that, but is it really more that the infra node maintains electrical power and just keeps running?

Reply:
Sorry, I am now more carefully reading earlier conversation and this was already answered at https://github.com/openshift/enhancements/pull/1005#discussion_r785201582